src/bvals/bvals.hpp

```diff
@@ -92,6 +92,62 @@ class BoundaryBase {
   static bool called_;
 };
 
+//----------------------------------------------------------------------------------------
+//! \class BoundarySwarms
+//  \brief centralized class for interacting with each individual swarm boundary data
+class BoundarySwarms : public BoundaryBase, BoundaryCommunication {
+ public:
+  BoundarySwarms(std::weak_ptr<MeshBlock> pmb, BoundaryFlag *input_bcs,
+                 ParameterInput *pin);
+
+  // variable-length arrays of references to all BoundarySwarm instances
+  std::vector<std::shared_ptr<BoundarySwarm>> bswarms;
```

Comment:
Why does this need to be a shared_ptr, can we not just make it a vector. This could improve iteration performance not that it will probably matter for this particular case because I don't image bswarms will get to large, but at least you won't be dealing with pointer indirection.

Reply:
I don't see any reason right now that you couldn't do that, but if this isn't a performance bottleneck now I'd rather maintain symmetry with `std::vector<std::shared_ptr<BoundaryVariable>> bvars;` (that is if these are changed I think they should both be changed together)

Reply:
ok